api/src/wfl/service/postgres.clj

```diff
@@ -67,13 +57,22 @@
            (remove (comp finished? :status))
            (run! #(update! % (get-status! workload %)))))))
 
-(def update-workflow-statuses!
-  "Use `tx` to update `status` of Cromwell `workflows` in a `workload`."
-  (letfn [(get-cromwell-status [{:keys [cromwell]} {:keys [uuid]}]
-            (if (util/uuid-nil? uuid)
-              "skipped"
-              (cromwell-status cromwell uuid)))]
-    (make-update-workflows get-cromwell-status)))
+(defn update-workflow-statuses!
+  "Use `tx` to update `status` of unfinished Cromwell `workflows` in a `workload`."
+  [tx {:keys [cromwell workflows] :as workload}]
+  (let [ws-to-update (->> workflows
+                          (remove (comp nil? :uuid))
+                          (remove (comp finished? :status)))
```

Comment:
the null UUID is used by the WGS pipeline for skipping workflows.

Reply:
> the null UUID is used by the WGS pipeline for skipping workflows.

OK, but that is not `uuid-nil`. That is `{… :uuid nil …}` or something.

Reply:
Filtering out nil UUIDs keeps us from querying cromwell with `[:id nil]`.  `make-update-workflows` does the same check in case the `get-status!` function it is passed can't handle `nil`

Reply:
Ah a nil uuid means it hasn't been submitted to cromwell yet. May be worth double checking that the main `update-workload!` implementation no-ops for workloads that are not `:started`

Reply:
Just for posterity's/documentation's sake: Ed is correct here, https://github.com/broadinstitute/wfl/blob/main/api/src/wfl/service/postgres.clj#L83